elastic/integrations
# [watchguard_firebox] Initial release of the WatchGuard Firebox

## Proposed commit message

Create New integration package watchguard_firebox.

- Added a log data stream.
- Added data collection logic for log data stream.
- Added the ingest pipeline for log data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for log data stream.
- Added system test cases for log data stream.



## Checklist


## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/watchguard_firebox directory.
- Run the following command to run tests.
> elastic-package test
```
--- Test results for package: watchguard_firebox - START ---
╭────────────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME                                                                   │ RESULT │ TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ watchguard_firebox │             │ asset     │ dashboard watchguard_firebox-5fb58f2b-8720-4b57-89b6-0f727f0d260f is loaded │ PASS   │        882ns │
│ watchguard_firebox │             │ asset     │ search watchguard_firebox-0c03afcf-e39d-484a-8575-0630b7bb892a is loaded    │ PASS   │        308ns │
│ watchguard_firebox │             │ asset     │ search watchguard_firebox-19e90429-51c9-4f47-8889-baf56bdc7735 is loaded    │ PASS   │        237ns │
│ watchguard_firebox │             │ asset     │ search watchguard_firebox-5cbe635b-9d2e-4d14-b2d4-e7146c421eaf is loaded    │ PASS   │        195ns │
│ watchguard_firebox │             │ asset     │ search watchguard_firebox-9173ff94-4af2-4f6d-86bc-7123d0f335f5 is loaded    │ PASS   │        216ns │
│ watchguard_firebox │ log         │ asset     │ index_template logs-watchguard_firebox.log is loaded                        │ PASS   │        184ns │
│ watchguard_firebox │ log         │ asset     │ ingest_pipeline logs-watchguard_firebox.log-0.1.0 is loaded                 │ PASS   │        162ns │
╰────────────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: watchguard_firebox - END   ---
Done
--- Test results for package: watchguard_firebox - START ---
╭────────────────────┬─────────────┬───────────┬────────────────────────────┬────────┬──────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME                  │ RESULT │ TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼────────────────────────────┼────────┼──────────────┤
│ watchguard_firebox │ log         │ pipeline  │ test-alarm.log             │ PASS   │  62.374853ms │
│ watchguard_firebox │ log         │ pipeline  │ test-diagnostic.log        │ PASS   │ 284.010886ms │
│ watchguard_firebox │ log         │ pipeline  │ test-event.log             │ PASS   │  153.26417ms │
│ watchguard_firebox │ log         │ pipeline  │ test-traffic.log           │ PASS   │ 230.823811ms │
│ watchguard_firebox │ log         │ pipeline  │ (ingest pipeline warnings) │ PASS   │ 264.588415ms │
╰────────────────────┴─────────────┴───────────┴────────────────────────────┴────────┴──────────────╯
--- Test results for package: watchguard_firebox - END   ---
Done
--- Test results for package: watchguard_firebox - START ---
╭────────────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ watchguard_firebox │ log         │ static    │ Verify sample_event.json │ PASS   │ 134.419749ms │
╰────────────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: watchguard_firebox - END   ---
Done
--- Test results for package: watchguard_firebox - START ---
╭────────────────────┬─────────────┬───────────┬───────────┬────────┬──────────────╮
│ PACKAGE            │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │ TIME ELAPSED │
├────────────────────┼─────────────┼───────────┼───────────┼────────┼──────────────┤
│ watchguard_firebox │ log         │ system    │ udp       │ PASS   │ 39.83772453s │
╰────────────────────┴─────────────┴───────────┴───────────┴────────┴──────────────╯
--- Test results for package: watchguard_firebox - END   ---
Done
```




## Related issues


- Closes #1203 

## Screenshots

![Integration Page](https://github.com/elastic/integrations/assets/123942796/31c3f9e7-6412-4fb3-bbaa-601d984e833f)
![Overview Page](https://github.com/elastic/integrations/assets/123942796/4809622d-30ce-415f-8bd3-39d6fd89b10d)
